### PR TITLE
Created EdgeDeviceGroup resource definition

### PR DESCRIPTION
**Pull request by jakub-dzon**
This PR introduces `EdgeDeviceGroup` resource for common device configuration.

### CODE REVIEW

File: api/v1alpha1/edgedevicegroup_types.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// EdgeDeviceGroupSpec defines the desired state of EdgeDeviceGroup
+type EdgeDeviceGroupSpec struct {
```
[the remark this answers is not in the document]
Comment (jakub-dzon) [reply api/v1alpha1/edgedevicegroup_types.go:27]: The design assumes that the devices have to choose one single group they belong to, hence no selector.

File: config/samples/management_v1alpha1_edgedevicegroup.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+apiVersion: management.project-flotta.io/v1alpha1
+kind: EdgeDeviceGroup
+metadata:
+  name: edgedevicegroup-sample
+spec:
+  # Add fields here
+  foo: bar
```
[the remark this answers is not in the document]
Comment (jakub-dzon) [reply config/samples/management_v1alpha1_edgedevicegroup.yaml:7]: You're right. That was auto-generated, never updated. Will do.

**Comment by gciavarrini** [comment]
Do you think it may be helpful to update the documentation (`crd.md`)?

**Comment by gciavarrini** [comment]
What about the controller? Will it be part of another PR?

**Review by gciavarrini — APPROVED** [review]
lgtm

**Review by masayag — APPROVED** [review]

**Pull request merged**